packages/client/test/end-to-end/ResendAll.test.ts

```diff
@@ -0,0 +1,121 @@
+import { StreamMessage } from 'streamr-client-protocol'
+import { ConfigTest, DOCKER_DEV_STORAGE_NODE } from '../../src/ConfigTest'
+import { Stream } from '../../src/Stream'
+import { StreamrClient } from '../../src/StreamrClient'
+import {
+    createTestStream,
+    describeRepeats,
+    fetchPrivateKeyWithGas,
+    getPublishTestStreamMessages,
+    getWaitForStorage
+} from '../test-utils/utils'
+
+const NUM_MESSAGES = 8
+const PARTITIONS = 3
+const WAIT_FOR_STORAGE_TIMEOUT = process.env.CI ? 20000 : 10000
+
+jest.setTimeout(60000)
+
+describeRepeats('ResendAll', () => {
+    let client: StreamrClient
+    let stream: Stream
+    let publishTestMessages: ReturnType<typeof getPublishTestStreamMessages>
+    let waitForStorage: (...args: any[]) => Promise<void>
+
+    // const createClient = getCreateClient()
+    beforeAll(async () => {
+        client = new StreamrClient({
+            ...ConfigTest,
+            auth: {
+                privateKey: await fetchPrivateKeyWithGas()
+            }
+        })
+
+        // eslint-disable-next-line require-atomic-updates
+        await client.connect()
+    })
+
+    beforeAll(async () => {
+        client.debug('createStream >>')
+        stream = await createTestStream(client, module, {
+            partitions: PARTITIONS,
+        })
+        client.debug('createStream <<')
+        client.debug('addToStorageNode >>')
+        await stream.addToStorageNode(DOCKER_DEV_STORAGE_NODE)
+        client.debug('addToStorageNode <<')
+
+        publishTestMessages = getPublishTestStreamMessages(client, stream.id)
+
+        waitForStorage = getWaitForStorage(client, {
+            stream,
+            timeout: WAIT_FOR_STORAGE_TIMEOUT,
+        })
+    }, WAIT_FOR_STORAGE_TIMEOUT * 2)
+
+    afterAll(async () => {
+        await client?.destroy()
+    })
+
+    describe('no data', () => {
```

Comment:
Maybe a comment here to let the future reader know that order of tests is important.

Reply:
sure, done.